### PR TITLE
feat: View annotations

### LINKED ISSUE #400: View Single Annotations
## Problem

Relates to https://github.com/gliff-ai/roadmap/issues/3.
Our users currently do not have a way to view a single assignees annotations on a single image.

## Solution

The user must have only one image selected. Two buttons appear under the metadata: `View Annotations` and `Compare Annotations` - each with their respective icon and tooltip.

View Annotations - Curate – Main Screen

Clicking the `View Annotations` button opens a dialogue for the user to select a user to view their annotations.

View Annotations - Curate – View Annotations Dialogue  open

The input field is a dropdown list (similar to other fields, this should have a search capability - i.e. user types a couple letters which filters the dropdown list, e.g. user input 'Jos' would only show 'Josh Smith' in the list.

View Annotations - Curate – View Annotations Dialogue  open w- dropdown list

The dropdown list should not show the user already selected in the list, i.e. 'Team Member 1' is selected already and thus not shown in the other list. Those users who are greyed out show they have not marked their annotations are complete yet (this message should also appear in a tooltip when user hovers - good user feedback practice). 

View Annotations - Curate – View Annotations Dialogue  open w- dropdown list – hover

User clicks the `View` button, and is taken to **ANNOTATE** (https://github.com/gliff-ai/annotate/issues/735).

## Considered Alternatives

N/A

### PR DESCRIPTION
**Pull request by philipjackson**
## Description

Adds the button / dialog for viewing annotations by other users, accessible only to team members / admins. This dialog will later be extended to allow selection of two users, to compare their annotations side-by-side.

closes #400 

## Checklist:

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request). If you're unsure about any of these, don't hesitate to leave a comment on this pull request!

- [ ] I have read the gliff.ai Contribution Guide.
- [ ] I have requested to **pull a branch** and not from main.
- [ ] I have checked all commit message styles match the requested structure.
- [ ] My code follows the style guidelines of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have performed a self-review of my own code.
- [ ] I have assigned **3 or less** reviewers.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.
- [ ] If appropriate, I have bumped any version numbers.

### CODE REVIEW

File: src/components/ViewAnnotationsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            sx={{
+              display: "flex",
+              marginTop: "20px",
+              justifyContent: "space-between",
+            }}
+          >
+            <BaseTextButton
+              id="cancel-view-annotations"
+              text="Cancel"
+              onClick={() => {}}
```
Comment (cooper667) [inline src/components/ViewAnnotationsDialog.tsx:88]: Should close dialog?

File: src/components/ViewAnnotationsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          size="small"
+          id="view-annotations"
+          onClick={() => {
+            //
+          }}
+        ></IconButton>
+      }
+    >
+      <Box sx={{ width: "400px" }}>
+        <CardContent>
```
Comment (cooper667) [inline src/components/ViewAnnotationsDialog.tsx:46]: Are these needed with our dialog component?

**Comment by ChasNelson1990** [comment]
Could we update the examples data to mock some assignees so there's a MWE of the dialog working?

File: src/components/ViewAnnotationsDialog.tsx (hunk 1 above)
Comment (philipjackson) [reply src/components/ViewAnnotationsDialog.tsx]: Fixed

**Comment by philipjackson** [comment]
> Could we update the examples data to mock some assignees so there's a MWE of the dialog working?

I've added some assignees to the example in `metadata.json`, so you can see them in the dialog (still won't redirect in standalone of course).

**Review by cooper667 — APPROVED** [review]

**Review by joshuajames-smith — CHANGES_REQUESTED** [review]
Screenshot 2022-05-27 at 10 29 06

Can we move the icon to below the metadata? And also replace the image with the `eye` icon to match the original design?

Screenshot 2022-05-27 at 10 30 07

The native browser password icon/functionality appears here and it shouldn't / doesn't anywhere else but the login/signup process? Can we remove this?

Screenshot 2022-05-27 at 10 31 08

There is a line/divider under the input field, can we remove this?

**Review by joshuajames-smith — CHANGES_REQUESTED** [review]
Screenshot 2022-05-31 at 12 44 44

Can we use `user name` rather than `user email`? Otherwise it looks good!

**Comment by philipjackson** [comment]
> Can we use `user name` rather than `user email`? Otherwise it looks good!

So I've got this working now, but just realised that names aren't necessarily unique, that's why we generally use emails as usernames. @ChasNelson1990 should we go with name names or stick to emails here?

**Comment by cooper667** [comment]
> > Can we use `user name` rather than `user email`? Otherwise it looks good!
> 
> So I've got this working now, but just realised that names aren't necessarily unique, that's why we generally use emails as usernames. @ChasNelson1990 should we go with name names or stick to emails here?

do users _have_ names?

**Comment by philipjackson** [comment]
> We ask for user `name` during sign up. This is likely to be much clearer UX when selecting users via name vs emails as not _all_ emails will be nice and clear.
> 
> I wonder now if it makes sense to change the `name` field within the sign up to `first name(s)` and `last name` so we ensure full user names are collected to be used by the platform?

So I had a look at how we currently handle this in MANAGE, seems we actually use both there, so I've done the same here for now :)

image

**Comment by cooper667** [comment]
>I wonder now if it makes sense to change the name field within the sign up to first name(s) and last name so we ensure full user names are collected to be used by the platform?

i don't want to do that migration 😀 


>So I had a look at how we currently handle this in MANAGE, seems we actually use both there, so I've done the same here for now :)

exactly what i was about to suggest

**Review by cooper667 — DISMISSED** [review]

**Review by ChasNelson1990 — DISMISSED** [review]

**Review by joshuajames-smith — CHANGES_REQUESTED** [review]
Screenshot 2022-06-08 at 10 27 39

^ I've noticed that the select field is automatically filled? I think we want to avoid this so users are required to select a user themselves. Can we ensure this select field is left empty to begin with?

Screenshot 2022-06-08 at 10 28 04

^ There is also another issue. If I select a user to view annotations on a first image and then cancel and select a second image to view annotations, the user appears the same. In the case above, that user is NOT assigned to that image so they shouldn't show anyway? I think the above fix would solve this anyways!

**Comment by philipjackson** [comment]
> ^ There is also another issue. If I select a user to view annotations on a first image and then cancel and select a second image to view annotations, the user appears the same. In the case above, that user is NOT assigned to that image so they shouldn't show anyway? I think the above fix would solve this anyways!

Good catch, it definitely wasn't supposed to do that. Fixed now :+1:

**Review by ChasNelson1990 — DISMISSED** [review]

**Review by joshuajames-smith — DISMISSED** [review]

**Review by cooper667 — DISMISSED** [review]

**Review by joshuajames-smith — APPROVED** [review]

**Review by SilviaZeta — APPROVED** [review]

**Pull request merged**